be/src/exec/pipeline/scan/scan_operator.cpp

```diff
@@ -141,7 +141,19 @@ bool ScanOperator::has_output() const {
         }
     }
 
-    return num_buffered_chunks() > 0;
+    const auto has_chunks = num_buffered_chunks() > 0;
+    // Avoid miss detach_chunk_source.
+    if (!has_chunks && _num_running_io_tasks == 0) {
```

Comment:
why the condition is   num_buffered_chunks() == 0 && _num_running_io_tasks == 0?

Reply:
> why the condition is num_buffered_chunks() == 0 && _num_running_io_tasks == 0?

Because only when there is no more chunks, and there is no more io_tasks, this scan_operator's task is done, we should detach_shared_input to notic is_finished to finish.

Reply:
> why the condition is num_buffered_chunks() == 0 && _num_running_io_tasks == 0?

This is unneeded, have removed.